lib/player.js

```diff
@@ -1413,25 +1477,286 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
         await this.unload(/* initializeMediaSource= */ false);
       }
       throw error;
+    } finally {
+      if (preloadManager) {
+        // This will cause any resources that were generated but not used to be
+        // properly destroyed or released.
+        await preloadManager.destroy();
+      }
+    }
+  }
+
+  /**
+   * Unloads the currently playing stream, if any, and returns a PreloadManager
+   * that contains the loaded manifest of that asset, if any.
+   * Allows for the asset to be re-loaded by this player faster, in the future.
+   * When in src= mode, this unloads but does not make a PreloadManager.
+   *
+   * @param {boolean=} initializeMediaSource
+   * @param {boolean=} keepAdManager
+   * @return {!Promise.<?shaka.media.PreloadManager>}
+   * @export
+   */
+  async unloadAndSavePreload(
```

Comment:
Maybe we need launch a error for src= or at least put a alwaysWarn, what do you think?